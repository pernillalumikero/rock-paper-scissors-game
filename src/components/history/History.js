import React from "react"

const History = ({history}) => {
  
  return (
    <>
      <ul id="ul">
        {history.length > 1 ? 
          history.map((item, index) =>{
          return (
          
            <li key={index}>
              {item.winner === item.playerOne ?
              
                  <>
                    <b>
                    {item.playerOne}
                    {" "}
                    {item.player1Choice}
                    </b>
                  </> 
                  : <>
                    {item.playerOne}
                    {" "}
                    {item.player1Choice}
                    </> 

            } {" "}
              {item.winner === item.playerTwo ?
              <>
                    <b>
                    {item.playerTwo}
                    {" "}
                    {item.player2Choice}
                    </b>
              </> 
                  : <>
                    {item.playerTwo}
                    {" "}
                    {item.player2Choice}
                    </>
              }
          </li>
          )
        }) : null}
      </ul>
    </>
  )
}

export default History
