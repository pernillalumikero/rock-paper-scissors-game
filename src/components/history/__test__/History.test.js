
import History from "../History";
import { render, screen } from "@testing-library/react";

describe("History", () => {

    test("Should render History", () => {

        const history = [{
            winner: "John",
            playerOne: "John",
            player1Choice: "rock",
            playerTwo: "Jane",
            player2Choice: "paper"
          }, 
          {
            winner: "Jane",
            playerOne: "John",
            player1Choice: "rock",
            playerTwo: "Jane",
            player2Choice: "paper"
          }];

        const tree = render(<History history={history} />).asFragment();
        expect(tree).toMatchSnapshot()
        screen.debug()
        

    })

    test("Should only show 10 lines and winner presented with bold text", () => {

        const history = new Array(19).splice(0,9).fill({
            winner: "playerOne",
            playerOne: "John",
            player1Choice: "rock",
            playerTwo: "Jane",
            player2Choice: "paper"
          });
          
          const {queryAllByRole} = render(<History history={history}/>)
          const historyItems = queryAllByRole("listitem")
          expect(historyItems.length).toBeLessThanOrEqual(10)

    })
    test("Should display winner with bold text", () => {

            const {queryAllByRole} = render(<History history={history}/>)
            const historyItems = queryAllByRole("listitem")
            historyItems.forEach((item) => {
            const winner = item.querySelector("b")
            expect(winner).toBeTruthy
        })
    })
})
