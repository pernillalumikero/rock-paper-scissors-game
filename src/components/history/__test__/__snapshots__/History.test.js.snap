// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History Should render History 1`] = `
<DocumentFragment>
  <ul
    id="ul"
  >
    <li>
      <b>
        John rock
      </b>
        Jane paper
    </li>
    <li>
      John rock  
      <b>
        Jane paper
      </b>
    </li>
  </ul>
</DocumentFragment>
`;
